forest/navigate.py

```diff
@@ -14,37 +14,54 @@
         rdt)
 
 
-class Config(object):
-    """File system navigation using config file
+class Navigator:
+    def __init__(self, config, database):
+        # TODO: Once the idea of a "Group" exists we can avoid using the
+        # config and defer the sub-navigator creation to each of the
+        # groups. This will remove the need for the `_from_group` helper
+        # and the logic in FileSystemNavigator.from_file_type().
+        # Also, it'd be good to switch the identification of groups from
+        # using the `pattern` to using the `label`. In general, not every
+        # group would have a `pattern`.
+        # e.g.
+        # self._navigators = {group.label: group.navigator for group in ...}
+        self._navigators = {group.pattern: self._from_group(group, database)
+                           for group in config.file_groups}
```

Comment:
I'm very happy with this comment, it is a good pointer for future developers to remember that we intend to change how navigators work